### ISSUE TITLE
Enable caching of additional task types

### ISSUE THREAD

**Issue by adarob** [google-research/text-to-text-transfer-transformer#293]
e.g., TextLineTask (#282)

**Comment by agemagician** [comment]
This is awesome, thanks a lot @adarob .
I was just going to open a reopen issue #282  because I couldn't integrate TFDS with the T5 Tasks.

**Comment by agemagician** [comment]
I have tested this commit, and the dataset generation function must include now a new parameter "shuffle" in order to work.
Old:
`def xxx_dataset_fn(split, shuffle_files=True):`
New:
`def xxx_dataset_fn(split, shuffle_files=True, shuffle=False):`

**Comment by agemagician** [comment]
Hmm, seems to be it doesn't support multithread, even after changing the parameters:
```
python -m t5.data.cache_tasks_main \
 --tasks=unsupervised_bio_denoising_task \
 --output_cache_dir=xxxxxxx \
 --module_import=bio_task \
--pipeline_options=["--runner=DirectRunner","--direct_num_workers 40"]
```

**Comment by adarob** [comment]
I just fixed the shuffle issue you pointed out.

Currently, this only parallelizes based on how you've sharded the inputs. So if you have 10 text files, it will run on 10 workers in parallel. I could make it split it out the line by line, but there is more overhead and it's a bigger refactor. Let me see if there is an elegant way to do it.

**Comment by agemagician** [comment]
Great, thanks for the shuffle fix.

hmmm, in my case, I have 6k files, and it runs a single thread:

```
trainFilesPath = tf.io.gfile.glob('gs://xxxx/train_*')
validFilesPath = tf.io.gfile.glob('gs://xxxxx/valid_*')

dataset_path = {
    "train": trainFilesPath,
    "validation": validFilesPath
}

def protein_dataset_fn(split, shuffle_files=True, shuffle=False):

  if shuffle_files == True and dataset_path[split] is list:
    random.shuffle(dataset_path[split])

  ds = tf.data.TextLineDataset(dataset_path[split]).skip(0)
  ds = ds.map(lambda *ex: dict(zip(["targets"], ex)))

  return ds
```

Did I miss something in my code ?

**Comment by adarob** [comment]
Ah yes, a generic task cannot be distributed, as mentioned in the comment. I could add a warning to this effect as well. If you use a TextLineTask, it will be distributed.
